rust/src/plutus.rs

```diff
@@ -976,34 +1009,56 @@ impl Deserialize for PlutusDataEnum {
 
 impl cbor_event::se::Serialize for PlutusData {
     fn serialize<'se, W: Write>(&self, serializer: &'se mut Serializer<W>) -> cbor_event::Result<&'se mut Serializer<W>> {
-        self.0.serialize(serializer)
+        match &self.original_bytes {
+            Some(bytes) => serializer.write_raw_bytes(bytes),
+            None => self.datum.serialize(serializer),
+        }
     }
 }
 
 impl Deserialize for PlutusData {
     fn deserialize<R: BufRead + Seek>(raw: &mut Deserializer<R>) -> Result<Self, DeserializeError> {
-        Ok(Self(PlutusDataEnum::deserialize(raw)?))
+        // these unwraps are fine since we're seeking the current position
+        let before = raw.as_mut_ref().seek(SeekFrom::Current(0)).unwrap();
+        let datum = PlutusDataEnum::deserialize(raw)?;
+        let after = raw.as_mut_ref().seek(SeekFrom::Current(0)).unwrap();
+        let bytes_read = (after - before) as usize;
+        raw.as_mut_ref().seek(SeekFrom::Start(before)).unwrap();
+        // these unwraps are fine since we read the above already
+        let original_bytes = raw.as_mut_ref().fill_buf().unwrap()[..bytes_read].to_vec();
+        raw.as_mut_ref().consume(bytes_read);
+        Ok(Self {
+            datum,
+            original_bytes: Some(original_bytes),
+        })
     }
 }
 
 impl cbor_event::se::Serialize for PlutusList {
     fn serialize<'se, W: Write>(&self, serializer: &'se mut Serializer<W>) -> cbor_event::Result<&'se mut Serializer<W>> {
-        if self.0.len() == 0 {
-          return Ok(serializer.write_array(cbor_event::Len::Len(0))?);
+        let use_definite_encoding = match self.definite_encoding {
```

Comment:
I decided to, at least right now, leave the adaption from #228's behavior in. I think it was breaking a test or two otherwise. One argument (besides saving 1 byte for non-empty lists) could be that it would cause things to break right away to stop people from making assumptions and not respecting the given byte encodings since lists are pretty fundamental and would be noticed early on (which happened in #227), instead of having it be noticed later on by some other inequality in some other part of code between us/cardano cli or even some other tool. There are other areas that it could vary like even with integer encodings or key orderings.

Should we keep this as changed in #228, or just always use definite like before?